### PR TITLE
feat: add edge unit tests to the docs

### PR DESCRIPTION
**Pull request by mansueli**
## What kind of change does this PR introduce?

Add unit tests docs for Edge Functions

## What is the current behavior?

Please link any relevant issues here.

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: apps/docs/pages/guides/functions/unit-test.mdx [hunk 1]
```diff
@@ -0,0 +1,132 @@
+import Layout from '~/layouts/DefaultGuideLayout'
+
+export const meta = {
+  id: 'unit-test',
+  title: 'Unit Tests for Edge Functions',
+  description: 'Unit Tests for Edge Functions with Deno Test',
+}
+To ensure the correctness and performance of your Edge Functions, testing is an essential step in the development process. Let's discuss the significance of testing and examining the code in the `deno-test.ts` file, which demonstrates how to test the Supabase client and the `hello-world` function.
```
Comment (laktek) [inline apps/docs/pages/guides/functions/unit-test.mdx:8]: Maybe worth mentioning, in which directory the user should create the test file and how to name it.
(I think it's better to place them in `supabase/tests` and can name the file by function name followed by `-test.ts` eg: hello-world-test.ts`)

File: apps/docs/pages/guides/functions/unit-test.mdx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+- We import various testing functions from the Deno standard library, including `assert`, `assertExists`, and `assertEquals`.
+- We import the `createClient` and `SupabaseClient` classes from the `@supabase/supabase-js` library to interact with the Supabase client.
+- We define the necessary configuration for the Supabase client, including the Supabase URL, API key, and authentication options.
+- The `testClientCreation` function tests the creation of a Supabase client instance and queries the database for data from a table. It verifies that data is returned from the query.
+- The `testHelloWorld` function tests the "Hello-world" Edge Function by invoking it using the Supabase client's `functions.invoke` method. It checks if the response message matches the expected greeting.
+- We run the tests using the `Deno.test` function, providing a descriptive name for each test case and the corresponding test function.
+
+Note: Please make sure to replace the placeholders (`supabaseUrl`, `supabaseKey`, `my_table`) with the actual values relevant to your Supabase setup.
```
Comment (gregnr) [inline apps/docs/pages/guides/functions/unit-test.mdx:87]: Worth making an `<Admonition>`?

**Review by gregnr — APPROVED** [review]

File: apps/docs/pages/guides/functions/unit-test.mdx [hunk 3]
```diff
@@ -0,0 +1,132 @@
+import Layout from '~/layouts/DefaultGuideLayout'
+
+export const meta = {
+  id: 'unit-test',
+  title: 'Unit Tests for Edge Functions',
+  description: 'Unit Tests for Edge Functions with Deno Test',
```
Comment (kiwicopple) [inline apps/docs/pages/guides/functions/unit-test.mdx:6]:
```suggestion
  title: 'Testing your Edge Functions',
  description: 'Writing Unit Tests for Edge Functions using Deno Test',
```

File: apps/docs/pages/guides/functions/unit-test.mdx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+}
+To ensure the correctness and performance of your Edge Functions, testing is an essential step in the development process. Let's discuss the significance of testing and examining the code in the `deno-test.ts` file, which demonstrates how to test the Supabase client and the `hello-world` function. When creating test files for Edge Functions, it is recommended to place them in the `supabase/tests` directory and name the file by the function name followed by `-test.ts`. For example, `hello-world-test.ts`.
```
Comment (kiwicopple) [inline apps/docs/pages/guides/functions/unit-test.mdx:8]:
```suggestion
}

To ensure the correctness and performance of your Edge Functions, testing is an essential step in the development process. Let's discuss the significance of testing and examining the code in the `deno-test.ts` file, which demonstrates how to test the Supabase client and the `hello-world` function. When creating test files for Edge Functions, it is recommended to place them in the `supabase/tests` directory and name the file by the function name followed by `-test.ts`. For example, `hello-world-test.ts`.
```

Comment (mansueli) [reply apps/docs/pages/guides/functions/unit-test.mdx:8]: This uses a test folder inside /functions/ subdirectory to create the test functions. So, I shouldn't interfere with: https://supabase.com/docs/guides/database/testing
